lib/engine/part/revenue_center.rb

```diff
@@ -50,9 +50,25 @@ def max_revenue
       end
 
       def route_revenue(phase, train)
-        return @revenue[:diesel] if train.name.upcase == 'D' && @revenue[:diesel]
+        revenue_multiplier(train) * route_base_revenue(phase, train)
+      end
+
+      def route_base_revenue(phase, train)
+        return (@revenue[:diesel]) if train.name.upcase == 'D' && @revenue[:diesel]
+
+        phase.tiles.reverse.each { |color| return (@revenue[color]) if @revenue[color] }
+      end
+
+      def revenue_multiplier(train)
+        distance = train.distance
+        base_multiplier = train.multiplier || 1
+
+        return base_multiplier if distance.is_a?(Numeric)
 
-        phase.tiles.reverse.each { |color| return @revenue[color] if @revenue[color] }
+        row = distance.index do |h|
+          h['nodes'].include?(type)
+        end
+        distance[row].fetch('multiplier', base_multiplier)
```

Comment:
why does the train and the distance both have multiplier?

Reply:
The train multiplier allows a basic "multiply all stop types" implementation. See 18MS trains.

The distance multiplier allows for multiplying specific nodes. 18AL/18Mex only double cities.